src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts

```diff
@@ -68,94 +73,147 @@ export class TechRecordSummaryComponent implements OnInit {
+      : (this.techRecordCalculated = { ...this.techRecord });
 
-    this.technicalRecordService.updateEditingTechRecord(this.vehicleTechRecordCalculated, true);
+    this.technicalRecordService.updateEditingTechRecord(this.techRecordCalculated, true);
   }
 
   handleFormState(event: any): void {
-    this.vehicleTechRecordCalculated = cloneDeep(this.vehicleTechRecordCalculated);
+    this.techRecordCalculated = cloneDeep(this.techRecordCalculated);
 
-    if (event.axles && event.axles.length < this.vehicleTechRecordCalculated.axles!.length) {
+    if (event.axles && event.axles.length < this.techRecordCalculated.axles!.length) {
       this.removeAxle(event);
-    } else if (event.axles && this.vehicleTechRecordCalculated.axles!.length < event.axles.length) {
+    } else if (event.axles && this.techRecordCalculated.axles!.length < event.axles.length) {
       this.addAxle(event);
     } else {
-      this.vehicleTechRecordCalculated = merge(this.vehicleTechRecordCalculated, event);
+      this.techRecordCalculated = merge(this.techRecordCalculated, event);
     }
 
-    if (event.brakes?.dtpNumber && event.brakes.dtpNumber.length >= 4 && this.vehicleTechRecordCalculated.vehicleType === VehicleTypes.PSV) {
-      this.setBodyFields(this.vehicleTechRecordCalculated.vehicleType);
+    if (event.brakes?.dtpNumber && event.brakes.dtpNumber.length >= 4 && this.techRecordCalculated.vehicleType === VehicleTypes.PSV) {
+      this.setBodyFields(this.techRecordCalculated.vehicleType);
     }
 
-    if (this.vehicleTechRecordCalculated.vehicleType === VehicleTypes.PSV && (event.grossKerbWeight || event.grossLadenWeight || event.brakes)) {
+    if (this.techRecordCalculated.vehicleType === VehicleTypes.PSV && (event.grossKerbWeight || event.grossLadenWeight || event.brakes)) {
       this.setBrakesForces();
     }
 
     if (
-      this.vehicleTechRecord.vehicleType === VehicleTypes.PSV ||
-      this.vehicleTechRecord.vehicleType === VehicleTypes.HGV ||
-      this.vehicleTechRecord.vehicleType === VehicleTypes.TRL
+      this.techRecord.vehicleType === VehicleTypes.PSV ||
+      this.techRecord.vehicleType === VehicleTypes.HGV ||
+      this.techRecord.vehicleType === VehicleTypes.TRL
     ) {
-      this.vehicleTechRecordCalculated.noOfAxles = this.vehicleTechRecordCalculated.axles?.length ?? 0;
+      this.techRecordCalculated.noOfAxles = this.techRecordCalculated.axles?.length ?? 0;
     }
 
-    this.technicalRecordService.updateEditingTechRecord(this.vehicleTechRecordCalculated);
-    this.formChange.emit();
+    this.technicalRecordService.updateEditingTechRecord(this.techRecordCalculated);
+
+    this.checkForms();
   }
 
-  generateAxleSpacing(numberOfAxles: number, setOriginalValues: boolean = false, axleSpacingOriginal?: AxleSpacing[]): AxleSpacing[] {
-    let axleSpacing: AxleSpacing[] = [];
+  checkForms(): void {
+    const forms = this.sections?.map(section => section.form).concat(this.customSectionForms);
```

Comment:
`this.sections` is of type `QueryList<DynamicFormGroupComponent>`, I'm not sure there is a `?` after `this.sections` here.

Of course, the correct implementation would be to remove the `!` on line 32 and make it `QueryList<DynamicFormGroupComponent> | undefined`, in which case the `?` makes sense. The reason I'm not suggesting using a `!` is because the query are ran before the `ngAfterViewInit` lifecycle hook, meaning it's not always defined, at constructor time for instance.

Reply:
Because it is a `ViewChild` and not an `Input`, I did not want to force `!` on it. You can enforce inputs by adding them to the selector but viewchilds do not have this advantage. E.g. if the element is dynamic on the template and it is removed by an `*ngIf` at some point, the viewchild will be `null`.